shell/platform/linux/public/flutter_linux/fl_pixel_buffer_texture.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_PIXEL_BUFFER_TEXTURE_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_PIXEL_BUFFER_TEXTURE_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include "fl_texture.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlPixelBufferTexture,
+                         fl_pixel_buffer_texture,
+                         FL,
+                         PIXEL_BUFFER_TEXTURE,
+                         FlTexture)
+
+struct _FlPixelBufferTextureClass {
+  FlTextureClass parent_class;
+
+  /**
+   * Virtual method called when Flutter wants to copy pixel buffer for
+   * populating this texture.
+   * @texture: an #FlPixelBufferTexture.
+   * @buffer: (out): pixel data.
+   * @format: (out): pixel buffer format (example GL_RGBA).
```

Comment:
Is it a good idea to have a `PixelBufferTexture`, which by itself seems quite agnostic about rendering backend use GL specific pixel buffer format? I think this should be either abstracted away, or removed completely and force single format. (i.e. pixel buffer on Windows expects RGBA).